### PR TITLE
finish ROS-330

### PR DESCRIPTION
**Pull request by AbnerZheng**
## Overview
Create primitive definitions located in Tuple.cc Part 3: 
  ./src/Tuple.cc:412:DEF("null?",obNullQ, 1, 1) **

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/ROS-330

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+import coop.rchain.rosette.{Ctxt, Fixnum, Ob, RblBool, Tuple}
 
 object tuple {
 
+  object obNullQ extends Prim {
+    override val name: String = "null?"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    override def fnSimple(ctxt: Ctxt): Right[PrimError, RblBool] = {
```
Comment (DoSOfRedRiver) [inline roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala]: Why don't use `checkArgumentMismatch` annotation to check if `argvec` actually contains an argument?

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala [hunk 2]
```diff
@@ -22,6 +22,15 @@ class TupleSpec extends FlatSpec with Matchers {
...
   )
 
+  /** null? */
+  "obNullQ" should "return true when first argument is tuple.NIL, otherwise return false" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Tuple.NIL))
+    obNullQ.fnSimple(newCtxt) should be(Right(RblBool(true)))
+
+    val newCtxt2 = ctxt.copy(nargs = 1, argvec = Tuple(Tuple(Fixnum(1))))
+    obNullQ.fnSimple(newCtxt2) should be(Right(RblBool(false)))
+  }
```
Comment (DoSOfRedRiver) [inline roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Please add a test case when `argvec` does not contain an argument.

File: roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala (hunk 1 above)
Comment (AbnerZheng) [reply roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala]: Done.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala (hunk 2 above)
Comment (AbnerZheng) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: When using the `checkArgumentMismatch` annotation, there is no need to create the test, am I right?

Comment (DoSOfRedRiver) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Yeah, I think so.

**Comment by AbnerZheng** [comment]
@DoSOfRedRiver I have added the checkArgumentMismatch annotation, PTAL.

**Comment by AbnerZheng** [comment]
@DoSOfRedRiver Could you please help me merge the PR?

**Pull request merged**